nilearn/image/resampling.py

```diff
@@ -441,11 +441,15 @@ def resample_img(img, target_affine=None, target_shape=None,
     affine = img.affine
 
     # noop cases
-    if target_affine is None and target_shape is None:
+    if (
+        np.shape(target_affine) == np.shape(affine)
+        and np.allclose(target_affine, affine)
+        and np.array_equal(target_shape, shape)
+    ):
         if copy and not input_img_is_string:
             img = _utils.copy_img(img)
         return img
-    if target_affine is affine and target_shape is shape:
+    if target_affine is affine and target_shape == shape:
```

Comment:
thanks! I think it is this line (452) that needs to change and the comparison to None above (l 444) could stay as it was

Reply:
You're right, of course. Perhaps you should push a commit of your own, right now I'm not comfortable with such details of the code, you actually wrote the patch.

Reply:
thanks for taking care of it! your last commit seems good!